server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/AggregationsIntegrationIT.java

```diff
@@ -39,25 +38,31 @@ public void setupSuiteScopeCluster() throws Exception {
 
     public void testScroll() {
         final int size = randomIntBetween(1, 4);
-        SearchResponse response = prepareSearch("index").setSize(size)
-            .setScroll(TimeValue.timeValueMinutes(1))
-            .addAggregation(terms("f").field("f"))
-            .get();
-        assertNoFailures(response);
-        Aggregations aggregations = response.getAggregations();
-        assertNotNull(aggregations);
-        Terms terms = aggregations.get("f");
-        assertEquals(Math.min(numDocs, 3L), terms.getBucketByKey("0").getDocCount());
-
-        int total = response.getHits().getHits().length;
-        while (response.getHits().getHits().length > 0) {
-            response = client().prepareSearchScroll(response.getScrollId()).setScroll(TimeValue.timeValueMinutes(1)).get();
-            assertNoFailures(response);
-            assertNull(response.getAggregations());
-            total += response.getHits().getHits().length;
-        }
-        clearScroll(response.getScrollId());
-        assertEquals(numDocs, total);
+        assertNoFailuresAndResponse(
+            prepareSearch("index").setSize(size).setScroll(TimeValue.timeValueMinutes(1)).addAggregation(terms("f").field("f")),
+            response -> {
+                Aggregations aggregations = response.getAggregations();
+                assertNotNull(aggregations);
+                Terms terms = aggregations.get("f");
+                assertEquals(Math.min(numDocs, 3L), terms.getBucketByKey("0").getDocCount());
+                final String[] scroll = new String[] { response.getScrollId() };
+                final int[] total = new int[] { response.getHits().getHits().length };
+                int currentTotal = 0;
+                while (total[0] - currentTotal > 0) {
```

Comment:
NIT: I think we could make this less nested by creating the arrays outside of `assertNoFailuresAndResponse`, then we can run the while loop at the top level of this method. This is fine too though, just a suggestion :)

Reply:
Yes, you are right, I wrote it less nested.